lib/src/facts/solaris/virtualization_resolver.cc

```diff
@@ -18,6 +18,17 @@ namespace facter { namespace facts { namespace solaris {
 
     string virtualization_resolver::get_hypervisor(collection& facts)
     {
+        auto isa = facts.get<string_value>(fact::hardware_isa);
+        if (isa && isa->value() == "sparc") {
+            auto ldom_domainrole_control = facts.get<boolean_value>("ldom_domainrole_control");
```

Comment:
It looks like ldom_domainrole_control will only be defined on sparc, so I can probably drop the hardware_isa checks.